tools/virt-host-validate-ch.c

```diff
@@ -0,0 +1,129 @@
+#include <config.h>
+
+#include "virt-host-validate-ch.h"
+#include "virt-host-validate-common.h"
+#include "virarch.h"
+#include "virbitmap.h"
+
+int validateMinimumChVersion(char* ver_string)
+{
+	// expected ver_string example: "cloud-hypversor v0.8.1-<hash>"
+    int min_major_version = 0;
+    int min_minor_version = 9;
+    char *ptr = strtok(ver_string, " ");
+    char *last_tok = NULL;
+    while(ptr != NULL)
+	{
+        last_tok = ptr;
+		ptr = strtok(NULL, " ");
+	}
+    if (last_tok == NULL)
+        return -1;
+
+    // eliminate 'v'
+	last_tok++;
+    ptr = strtok(last_tok, ".");
+	if (ptr == NULL)
+		return -1;
+    int majorVer = atoi(ptr); 
+    ptr = strtok(NULL, ".");
+	if (ptr == NULL)
+		return -1;
+	int minorVer = atoi(ptr);
+
+	if (majorVer > min_major_version)
+	{
+		return 0;
+	}
+	else if (majorVer == min_major_version)
+	{
+		if (minorVer >= min_minor_version)
+			return 0;
+		else
+			return -1;
+	}
+	else
+	{
+		return -1;
+	}
+}
+
+int virHostValidateCh(void)
+{
+	int ret = 0;
+	virBitmapPtr flags;
+	bool hasHwVirt = false;
+	bool hasVirtFlag = false;
+	virArch arch = virArchFromHost();
+	const char *kvmhint = _("Check that CPU and firmware supports virtualization "
+			"and kvm module is loaded");
+
+	if (!(flags = virHostValidateGetCPUFlags()))
+		return -1;
+
+	switch ((int)arch) {
+		case VIR_ARCH_I686:
```

Comment:
I'm not sure about this arch check given the supported arch [list](https://github.com/cloud-hypervisor/cloud-hypervisor#architectures). It probably should just be checks on VIR_ARCH_X86_64 and VIR_ARCH_AARCH64.

Reply:
Ah I see the style issues also fixed up most of the arch checks sorry. Still I think the I686 test should be removed.

Reply:
Removed I686.